File: Code.java
String findProfession(int level, int pos) {
    
    String res = (pos%2 == 0) ? "Doctor" : "Engineer";
    
    for(int i = 1; i < level; i++)
    {
        pos = (int)Math.ceil(pos / 2.0);
        
        if(pos % 2 == 0)
        {
            
            if(res.equals("Doctor"))
                res = "Engineer";
            else
                res = "Doctor";
        }
    }

    return res;
}

File: Code.c
char * findProfession(int level, int pos) {
    
    bool isEng = (pos%2 == 0) ? false : true;
        
    for(int i = 1; i < level; i++){
        pos = (int)ceil((double)pos/2);
        
        if(pos%2 == 0)
            if(isEng)
                isEng = false;
            else
                isEng = true;
    }
    
    return isEng ? "Engineer" : "Doctor";
}
